lib/ansible/galaxy/collection/__init__.py

```diff
@@ -586,6 +594,180 @@ def build_collection(collection_path, output_path, force):
+            to_text(dep, errors='surrogate_or_strict'), ', '.join(needed_by))
+        )
+
+
+def get_matching_collections(requirements, installed_collections, force):
+    """Gets the list of collections that match the requirements list
+
+    :param requirements: A list of collection specifications to look for in the list of collections
+    :param installed_collections: A list of CollectionRequirement instances
+    :param force: A boolean to bypass prompting when multiple matching instances are found
+    :return: A list of CollectionRequirement instances
+    """
+    matching_collections = []
+    for requirement in requirements:
+        unfiltered_matches = []
+
+        req_name = requirement[0]
+        req_version = requirement[1]
+        b_req_name = to_bytes(req_name, errors='surrogate_or_strict')
+
+        if os.path.isfile(b_req_name) or urlparse(req_name).scheme.lower() in ['http', 'https'] or len(req_name.split('.')) != 2:
+            display.display("Skipping '%s' because it is not a valid collection name. The format namespace.name is expected." % req_name)
+
+        for installed_collection in installed_collections:
+            if to_text(installed_collection) == req_name:
+                try:
+                    installed_collection.add_requirement(None, req_version)
+                except AnsibleError:
+                    continue
+                else:
+                    unfiltered_matches.append(installed_collection)
+
+        matching_collections.extend(
+            filter_multiple_matches('%s:%s' % (req_name, req_version), unfiltered_matches, force)
+        )
+    return matching_collections
+
+
+def filter_multiple_matches(req, collections, force):
+    """Filter multiple collections found for a requirement by prompting the user
+
+    :param req: The 'namespace.name:version' string for the prompt
+    :param collections: A list of CollectionRequirement instances
+    :param force: A boolean to bypass prompting and return all collections without filtering
+    :return: A list of installed CollectionRequirement instances
+    """
+    if len(collections) <= 1 or force:
+        return collections
+
+    indices = list(range(0, len(collections)))
+
+    prompt = "Found the following collections that match '%s'" % req
+    for num in indices:
+        prompt += "\n\t%s. (%s) %s" % (num, collections[num].latest_version, to_text(collections[num].b_path, errors='surrogate_or_strict'))
+    prompt += "\nType the comma separate list of numbers to remove (a for all or n for none): "
+
+    while True:
+        result = display.prompt(prompt)
+        if result in ['a', 'all']:
+            break
+        elif result in ['n', 'none']:
+            indices = []
+            break
+        else:
+            try:
+                indices = [int(r) if int(r) in indices else int('invalid') for r in result.split(',')]
+            except ValueError:
+                pass
+            else:
+                break
```

Comment:
Nesting is unnecessary here
```suggestion
        try:
            indices = [int(r) if int(r) in indices else int('invalid') for r in result.split(',')]
        except ValueError:
            pass
        else:
            break
```